include/robot_interfaces/n_joint_robot_types.hpp

```diff
@@ -43,6 +50,42 @@ struct NJointRobotTypes
         //! D-gain for position controller.  If NaN, default is used.
         Vector position_kd;
 
+        std::vector<std::string> get_name() override
+        {
+          std::vector<std::string> names;
+          names.push_back("torque");
+          names.push_back("position");
+          names.push_back("position_kp");
+          names.push_back("position_kd");
+          return names;
+        }
+
+        std::vector<std::vector<double>> get_data() override
+        {
+          //first map the Eigen vectors to std::vectors
+          std::vector<double> torque_;
+          torque_.resize(torque.size());
+          Vector::Map(&torque_[0], torque.size()) = torque;
+
+          std::vector<double> position_;
+          position_.resize(position.size());
+          Vector::Map(&position_[0], position.size()) = position;
+
+          std::vector<double> position_kp_;
+          position_kp_.resize(position_kp.size());
+          Vector::Map(&position_kp_[0], position_kp.size()) = position_kp;
+
+          std::vector<double> position_kd_;
+          position_kd_.resize(position_kd.size());
+          Vector::Map(&position_kd_[0], position_kd.size()) = position_kd;
+
+          //then return them in a fixed size vector of vectors to avoid copying due to pushing back
+          //value of information!
+          std::vector<std::vector<double>> result;
+          result = {torque_, position_, position_kp_, position_kd_};
```

Comment:
This actually results in a copy (although the compiler may be smart enough to optimize it away).
To avoid it, you can first create the `result` vector with uninitialized entries of the required size and then write the values directly into it:
```
std::vector<std::vector<double>> result = {std::vector<double>(N), std::vector<double>(N), std::vector<double>(N), std::vector<double>(N)};
Vector::Map(&result[0][0], torque.size()) = torque;
```
(not tested but I think it should work like this).

Reply:
Great, thank you!